src/xref2/ref_tools.ml

```diff
@@ -96,19 +96,40 @@ let ambiguous_ref_warning name results =
     (Format.pp_print_list ~pp_sep pp_kind)
     results
 
+let ambiguous_label_warning name (labels : Component.Element.any list) =
+  (* Sort the results to make sure the result is reproducible. *)
+  let pp_kind pp r =
+    match r with
+    | `Label (_, l) ->
+        Odoc_model.Location_.pp_span_start pp l.Component.Label.location
+    | _ -> ()
+  in
+  Lookup_failures.report_warning
+    "@[<2>Multiple sections named '%s' found. Please alter one to ensure \
+     reference is unambiguous. Locations:@ %a@]"
+    name
+    (Format.pp_print_list ~pp_sep:Format.pp_force_newline pp_kind)
+    labels
+
+let ambiguous_warning name (results : [< Component.Element.any ] list) =
+  let results = (results :> Component.Element.any list) in
+  if List.for_all (function `Label _ -> true | _ -> false) results then
+    ambiguous_label_warning name (results :> [> Component.Element.any ] list)
```

Comment:
This cast is unnecessary.

Reply:
Ah yes! thanks, this was left in from when I was trying to return a different `` `AmbiguousLabel `` error (which ultimately failed, predictably in hindsight).